### PR TITLE
Updating Service Bus Templates

### PR DESCRIPTION
**Pull request by wsilveiranz**
- Adding sequential convoy template
- Adding category to all service bus template (design patterns)

### CODE REVIEW

**Comment by wsilveiranz** [comment]
@divyaswarnkar, @Elaina-Lee please review content for technical accuracy.

File: sequential-convoy/manifest.json [hunk 1]
```diff
@@ -0,0 +1,60 @@
+{
+    "title": "Service Bus Peek Lock Pattern - Renew lock token",
+    "description": "Get new message from a Service Bus queue and lock it for processing, renewing the lock token until processing is completed. The message will be deleted from the queue after the processing is completed. This pattern is useful when you want to ensure that the message is processed only once and not lost in case of a failure during processing.",
```
Comment (ecfan) [inline sequential-convoy/manifest.json:3]: Consistency edits to mirror descriptions in other templates:

```suggestion
    "description": "This workflow template triggers when a message is available in a Service Bus queue. The workflow gets the message from the queue, locks the message for processing, and renews the lock token until processing completes. The message is deleted from the queue after processing completes. This pattern helps you make sure that the workflow processes the message only once and that the message doesn't get lost if a failure happens during processing.",

```

File: sequential-convoy/manifest.json [hunk 2]
```diff
@@ -0,0 +1,60 @@
+{
+    "title": "Service Bus Peek Lock Pattern - Renew lock token",
```
Comment (ecfan) [inline sequential-convoy/manifest.json:2]: Consistency with other templates:

```suggestion
    "title": "Azure Service Bus: Get message from a queue with token renewal - Peek lock renew pattern",

```

File: sequential-convoy/manifest.json [hunk 3]
```diff
@@ -0,0 +1,60 @@
+{
+    "title": "Service Bus Peek Lock Pattern - Renew lock token",
+    "description": "Get new message from a Service Bus queue and lock it for processing, renewing the lock token until processing is completed. The message will be deleted from the queue after the processing is completed. This pattern is useful when you want to ensure that the message is processed only once and not lost in case of a failure during processing.",
+    "prerequisites": "Before you can use this template, you need to configure a Service Bus namespace and create a queue. For more information about Service Bus see [Create a new queue on Service Bus](https://learn.microsoft.com/en-us/azure/service-bus-messaging/service-bus-quickstart-portal)",
```
Comment (ecfan) [inline sequential-convoy/manifest.json:4]: - Consistency edits with other templates.
-  We shouldn't hardcode URLs, nor include the locale (en-us) in the link, so we need to replace with an FW link.

```suggestion
    "prerequisites": "Before you can use this template, you need to create or set up a Service Bus namespace with a queue. For more information, see [Create a Service Bus namespace and a queue](https://go.microsoft.com/fwlink/?linkid=2296007).",

```

File: sequential-convoy/manifest.json [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        }
+    ],
+    "images": {
+        "light": "snapshot_light",
+        "dark": "snapshot_dark"
+    },
+    "parameters": [
+        {
+            "name": "queueName_#workflowname#",
+            "displayName": "Queue Name",
```
Comment (ecfan) [inline sequential-convoy/manifest.json:31]: - Use sentence casing for parameter names, like the other templates.

```suggestion
            "displayName": "Queue name",

```

File: sequential-convoy/manifest.json [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+            "type": "String",
+            "default": "This is the name of the queue from which the message will be fetched.",
```
Comment (ecfan) [inline sequential-convoy/manifest.json:33]:
```suggestion
            "default": "The name of the queue from where to get the message.",

```

File: sequential-convoy/manifest.json [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+            "description": "This is a parameter",
```
Comment (ecfan) [inline sequential-convoy/manifest.json:34]:
```suggestion
            "description": "The name of the Service Bus queue from where to get the message.",

```

File: sequential-convoy/manifest.json [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+            "required": true
+        },
+        {
+            "name": "messageBatchSize_#workflowname#",
+            "displayName": "Message Batch Size",
```
Comment (ecfan) [inline sequential-convoy/manifest.json:39]:
```suggestion
            "displayName": "Message batch size",

```

File: sequential-convoy/manifest.json [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+            "type": "Int",
+            "default": "50",
+            "description": "This indicates how many messages will be fetched in a single batch.",
```
Comment (ecfan) [inline sequential-convoy/manifest.json:42]:
```suggestion
            "description": "The number of messages to get as a single batch.",

```

File: sequential-convoy/manifest.json [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+            "required": true
+        },
+        {
+            "name": "delayInMinutes_#workflowname#",
+            "displayName": "Delay before next token renewal",
+            "type": "Int",
+            "default": "3",
+            "description": "This indicates how long to wait for processing before the lock token is renewed. This value should be smaller than the lock token duration of the queue.",
```
Comment (ecfan) [inline sequential-convoy/manifest.json:50]:
```suggestion
            "description": "The number of minutes to delay processing before renewing the lock token. Make sure this value is smaller than the queue's lock token duration.",

```

**Review by ecfan — CHANGES_REQUESTED** [review]
@wsilveiranz: Feedback to make UX content similar to other templates for consistency.

**Comment by ecfan** [comment]
#assign:ecfan

**Comment by wsilveiranz** [comment]
Thanks @ecfan - I've accepted all the suggestions.

**Comment by ecfan** [comment]
@wsilveiranz: Can we please change the folder to look like the others: 

get-message-service-bus-queue-peek-lock-renew-sequential-convoy

**Comment by wsilveiranz** [comment]
> @wsilveiranz: Can we please change the folder to look like the others:
> 
> get-message-service-bus-queue-peek-lock-renew-sequential-convoy

Done!

**Comment by ecfan** [comment]
[celebrate]     Esther Fan reacted to your message:
________________________________

File: get-message-service-bus-queue-peek-lock-renew-sequential-convoy/manifest.json [hunk 10]
```diff
@@ -0,0 +1,69 @@
+{
+    "title": "Azure Service Bus: Get message from a queue with token renewal - Peek lock renew pattern",
```
Comment (ecfan) [inline get-message-service-bus-queue-peek-lock-renew-sequential-convoy/manifest.json:2]:
```suggestion
    "title": "Azure Service Bus (sequential convoy): Get message from a queue with token renewal - Peek lock renew pattern",

```

Comment (ecfan) [reply get-message-service-bus-queue-peek-lock-renew-sequential-convoy/manifest.json:2]: This version could work too: 

"Azure Service Bus: Get message from a queue with token renewal - Peek lock renew pattern (sequential convoy)"

**Comment by ecfan** [comment]
[celebrate]     Esther Fan reacted to your message:
________________________________

File: get-message-service-bus-queue-peek-lock-renew-sequential-convoy/manifest.json [hunk 11]
```diff
@@ -1,9 +1,9 @@
 {
-    "title": "Azure Service Bus: Get message from a queue with token renewal - Peek lock renew pattern",
+    "title": "Azure Service Bus: Sequential convoy pattern",
 
-    "description": "This workflow template triggers when a message is available in a Service Bus queue. The workflow gets the message from the queue, locks the message for processing, and renews the lock token until processing completes. The message is deleted from the queue after processing completes. This pattern helps you make sure that the workflow processes the message only once and that the message doesn't get lost if a failure happens during processing.",
+    "description": "This workflow template triggers when a message is available in a Service Bus session enabled queue. The workflow processes related messages from a specific session on the same workflow instance, creating different workflow instances for each session in the queue.",
```
Comment (ecfan) [inline get-message-service-bus-queue-peek-lock-renew-sequential-convoy/manifest.json:4]: I'm having trouble parsing some of this. I'm not sure which specific item creates difference workflow instances for each session in the queue?

File: get-message-service-bus-queue-peek-lock-renew-sequential-convoy/manifest.json [hunk 12]
```diff
@@ -1,9 +1,9 @@
...
+    "description": "This workflow template triggers when a message is available in a Service Bus session enabled queue. The workflow processes related messages from a specific session on the same workflow instance, creating different workflow instances for each session in the queue.",
 
-    "prerequisites": "Before you can use this template, you need to create or set up a Service Bus namespace with a queue. For more information, see [Create a Service Bus namespace and a queue](https://go.microsoft.com/fwlink/?linkid=2296007).",
+    "prerequisites": "Before you can use this template, you need to configure a Service Bus namespace and create a queue with session enabled. For more information about Service Bus see [Create a new queue on Service Bus](https://learn.microsoft.com/en-us/azure/service-bus-messaging/service-bus-quickstart-portal). You also need to configure your Logic App to support stateful connectors, which is required for sessions. For more information see [Enable stateful mode for stateless built-in connectors in Azure Logic Apps](https://go.microsoft.com/fwlink/?linkid=2296010)",
```
Comment (ecfan) [inline get-message-service-bus-queue-peek-lock-renew-sequential-convoy/manifest.json:6]: @wsilveiranz: Was there a reason for not using the previously provided FW link? We shouldn't be using hardcoded doc URLs with the locale (en-us) included.

```suggestion
    "prerequisites": "Before you can use this template, you need to set up a Service Bus namespace and create a queue with sessions enabled. For more information, see [Create a Service Bus namespace and a queue](https://go.microsoft.com/fwlink/?linkid=2296007) and [Enable message sessions for a Service Bus queue](https://go.microsoft.com/fwlink/?linkid=2296202). You also need to set up your logic app to support stateful connectors, which is required for sessions. For more information, see [Enable stateful mode for stateless built-in connectors in Azure Logic Apps](https://go.microsoft.com/fwlink/?linkid=2296010).",

```

File: get-message-service-bus-queue-peek-lock-renew-sequential-convoy/manifest.json [hunk 13]
```diff
@@ -1,9 +1,9 @@
 {
-    "title": "Azure Service Bus: Get message from a queue with token renewal - Peek lock renew pattern",
+    "title": "Azure Service Bus: Sequential convoy pattern",
```
Comment (ecfan) [inline get-message-service-bus-queue-peek-lock-renew-sequential-convoy/manifest.json]: This title appears on the gallery browse page and should be more descriptive for customers who don't know what "sequential convoy" means as well as those who do. Also use word structure similar to other template titles (verb-first wording).

```suggestion
    "title": "Azure Service Bus: Process related messages from a session-enabled queue for the same workflow instance - Sequential convoy pattern",

```

**Review by ecfan — CHANGES_REQUESTED** [review]

File: get-message-service-bus-queue-peek-lock-renew-sequential-convoy/manifest.json (hunk 12 above)
Comment (wsilveiranz) [reply get-message-service-bus-queue-peek-lock-renew-sequential-convoy/manifest.json:6]: The first link was a copy and paste error.

**Comment by wsilveiranz** [comment]
@ecfan, I think some of my commits overrode yours, but I updated them now to align. I've also updated the description to try and clarify the process. Let me know if you are ok with that.

File: get-message-service-bus-queue-peek-lock-renew-sequential-convoy/manifest.json [hunk 14]
```diff
@@ -0,0 +1,70 @@
+{
+    "title": "Azure Service Bus: Process related messages from a session-enabled queue for the same workflow instance - Sequential convoy pattern",
+
+    "description": "This workflow template triggers when a message is available in a Service Bus session enabled queue. The workflow creates a new instances for each session available in the queue, processes related messages from that session in the same workflow instance, one at a time.",
```
[the remark this answers is not in the document]
Comment (wsilveiranz) [reply get-message-service-bus-queue-peek-lock-renew-sequential-convoy/manifest.json]: Exactly. You nailed it!

**Review by ecfan — APPROVED** [review]
Thanks!

**Review by ecfan — APPROVED** [review]

**Review by Elaina-Lee — APPROVED** [review]

**Pull request merged**